ambry-router/src/main/java/com/github/ambry/router/TtlUpdateOperation.java

```diff
@@ -355,6 +362,15 @@ private void onErrorResponse(ReplicaId replicaId, RouterException exception, boo
   private void checkAndMaybeComplete() {
     // operationCompleted is true if Blob_Authorization_Failure was received.
     if (operationTracker.isDone() || operationCompleted) {
+      // JING TODO next revision, will add configuration for this feature
+      if (!operationTracker.hasSucceeded()) {
+        // the operation is failed, try to recover with ReplicateBlob
+        // if retryWithReplicateBlob returns true, it's under retry. operation is not completed yet.
+        if (retryWithReplicateBlob()) {
```

Comment:
Retry should be done asynchronous, right. We still return 503 back to client and at the same time, we run the replicate blob so next time client comes back with the same ttlupdate request, it would be a success.